src/docks/encodedock.cpp

```diff
@@ -2042,3 +2047,38 @@ bool EncodeDock::detectHardwareEncoders()
     }
     return !hwlist.isEmpty();
 }
+
+bool EncodeDock::checkForMissingFiles()
+{
+    Mlt::Producer* service = fromProducer();
+    if (!service) {
+        service = MAIN.playlist();
+    }
+    if (!service) {
+        LOG_ERROR() << "Encode: No service to encode";
+        return true;
+    }
+    QTemporaryFile tmp;
+    tmp.open();
+    tmp.close();
+    QString fileName = tmp.fileName();
+    tmp.remove();
+    MLT.saveXML(fileName, service, false /* without relative paths */, nullptr, Settings.proxyEnabled());
```

Comment:
This is a different logic than lines 926 - 927:
```
    auto isProxy = ui->previewScaleCheckBox->isChecked() && Settings.proxyEnabled();
    MLT.saveXML(fileName, service, false /* without relative paths */, nullptr, isProxy);
```
Also, I do not see where this temporary file is removed after `MLT.saveXML()`

Reply:
>This is a different logic than lines 926 - 927
OK. I will change the logic to match.

> Also, I do not see where this temporary file is removed
"the file will subsequently be removed upon destruction of the QTemporaryFile object"
https://doc.qt.io/qt-5/qtemporaryfile.html

Reply:
I saw the `tmp.remove()` and thought it might invalidate the auto-removal, but I think you are correct.